index.js

```diff
@@ -56,13 +53,14 @@ class ProtomuxRpcClient extends ReadyResource {
 
   async close () {
     this._backoff.destroy()
-    this.rpc.destroy()
+    if (this.rpc) this.rpc.destroy()
     if (this._pendingRPC) this._pendingRPC.destroy()
     return super.close()
   }
 
   async _close () {
     if (this._connecting) await this._connecting // Debounce
+    this._suspendedSignal.notify() // flush any pending requests
```

Comment:
notify prop before the await above?

Reply:
I think they're equivalent:
- Current flow means any pending requests will early-return from `_makeRequest` after the _close completes
- notifying before the await means they can return from `_makeRequest` while _close is running

Reply:
So I'm fine with either